### PR TITLE
Fixes #10869, added {role} to the dir name that stores the workunits.list

### PR DESCRIPTION
**Pull request by yuriw**
@jdurgin Pls take a look and then I will send it to Zack

Signed-off-by: Yuri Weinstein <yuri.weinstein@inktank.com>

### CODE REVIEW

**Comment by yuriw** [comment]
@zmc   Pls review, it fixes http://tracker.ceph.com/issues/10869

File: tasks/workunit.py [hunk 1]
```diff
@@ -308,13 +308,13 @@ def _run_tests(ctx, refspec, role, tests, env, subdir=None, timeout=None):
             'if', 'test', '-e', 'Makefile', run.Raw(';'), 'then', 'make', run.Raw(';'), 'fi',
             run.Raw('&&'),
             'find', '-executable', '-type', 'f', '-printf', r'%P\0'.format(srcdir=srcdir),
-            run.Raw('>{tdir}/workunits.list'.format(tdir=testdir)),
+            run.Raw('>{tdir}/workunits.list.{role}'.format(tdir=testdir,role=role)),
```
Comment (zmc) [inline tasks/workunit.py]: Needs a space between args, e.g. ``format(tdir=testdir, role=role)``

File: tasks/workunit.py [hunk 2]
```diff
@@ -308,13 +308,13 @@ def _run_tests(ctx, refspec, role, tests, env, subdir=None, timeout=None):
...
         ],
     )
 
     workunits = sorted(misc.get_file(
         remote,
-        '{tdir}/workunits.list'.format(tdir=testdir)).split('\0'))
+        '{tdir}/workunits.list.{role}'.format(tdir=testdir,role=role)).split('\0'))
```
Comment (zmc) [inline tasks/workunit.py]: Needs a space between args, e.g. ``format(tdir=testdir, role=role)``

File: tasks/workunit.py [hunk 3]
```diff
@@ -368,6 +368,6 @@ def _run_tests(ctx, refspec, role, tests, env, subdir=None, timeout=None):
         remote.run(
             logger=log.getChild(role),
             args=[
-                'rm', '-rf', '--', '{tdir}/workunits.list'.format(tdir=testdir), srcdir,
+                'rm', '-rf', '--', '{tdir}/workunits.list.{role}'.format(tdir=testdir,role=role), srcdir,
```
Comment (zmc) [inline tasks/workunit.py]: Needs a space between args, e.g. ``format(tdir=testdir, role=role)``

**Comment by zmc** [comment]
Mainly style things here. I made a few identical comments that should be addressed. Also, the commit message is too long to fit in most interfaces - 50 chars max, and in a paragraph below that you can put all the detail you need.

Finally, are you sure you just want to put this into giant? It will never automatically get into master if that's the case.

**Comment by yuriw** [comment]
@zmc - I hope fixed all spaces.  Re: master, I was planning to do cherry-pick after this one is merged.  Do we use a different way?  If we do, I'd be happy to follow it.

**Comment by zmc** [comment]
@yuriw Thanks for making those changes.

Normally we commit to master and then cherry-pick from there; in the future let's stick to that but this time I think we can make an exception since I don't think it will actually hurt anything.

**Comment by yuriw** [comment]
@zmc cherry-picked to master 9942440ddf9f6847e76de3b49f1ad0d98c62a3d0

**Pull request merged**